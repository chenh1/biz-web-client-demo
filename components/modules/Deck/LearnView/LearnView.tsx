'use client'

import type { VideoType } from '@/graphql/queries/videos';
import React, { useState } from 'react';
import { Heading } from '../../../core/Heading/Heading';
import { VideoPlayer } from '../../../core/VideoPlayer/VideoPlayer';
import { Box } from '../../../containers/Box/Box';
import { AccordionDivider } from '../../../containers/AccordionDivider/AccordionDivider';
import { Overlay } from '../../../containers/Overlay/Overlay';
import { Button } from '../../../core/Button/Button';
import { useSession } from "next-auth/react";
import { EXAM } from '../Deck';
import { Markdown } from '../../../utils/Markdown';

interface LearnViewProps {
  video: VideoType;
  setMode: (string) => void;
}

export const LearnView: React.FC<LearnViewProps> = ({ video, setMode }: LearnViewProps) => {
  const [ videoEnded, setVideoEnded ] = useState<boolean>(false)
  const session = useSession();

  return (
    <>
      <Box padding="0" gap="md" className="place-items-center w-full mt-[50%] md:mt-[20%]">
        <Box padding="0" className="w-full place-self-center place-items-center">
          <Heading type="h2" fontSizeClass="text-xl sm:text-5xl" className="text-center">{video?.title}</Heading>
          <Box padding="0" className="justify-self-stretch sm:justify-self-center">
            <Button onClick={() => setMode(EXAM)} label={videoEnded ? "Test your knowledge!" : "Skip to quiz"} primary={videoEnded}/>
          </Box>
        </Box>
        <VideoPlayer 
          isResponsive 
          url={video?.url}
          onEnded={() => setVideoEnded(true)}
          controls
        />
        {/* <Button label="Add entry" onClick={() => addCompletedModulesEntries({ variables: { moduleData: [{ lessonModule: 1.05, examScore: 0.8 }] } })} />
        <Button label="Delete entry" onClick={() => deleteCompletedModulesEntries({ variables: { moduleData: [{ lessonModule: 1.05, examScore: 0.8 }] } })} />
        <Button label="Update entry" onClick={() => updateCompletedModulesEntries({ variables: { moduleData: [{ lessonModule: 1.05, examScore: 0.3 }] } })} /> */}
        {video?.description &&
          <Box padding="0" className="w-full">
            <AccordionDivider label={<Heading type="h4">Transcript</Heading>} dividerShouldExpand>
              <Markdown gap="md">{video?.transcript}</Markdown>
            </AccordionDivider>
          </Box>
        }
        {videoEnded &&
          <Overlay visibility={videoEnded} toggleVisibility={setVideoEnded}>
            <Box padding="0" className="w-full">
              <Heading type="h2">You've completed this video!</Heading>
              <Button onClick={() => setMode(EXAM)} label="Test your knowledge!" primary/>
            </Box>
          </Overlay>
        }
      </Box>
    </>
  )
}