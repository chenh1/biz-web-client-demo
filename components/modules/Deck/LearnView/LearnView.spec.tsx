import { LearnView } from "./LearnView";
import { fireEvent, render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window

jest.mock("react-markdown", () => ({ children }) => <div>{children}</div>);
jest.mock("rehype-raw", () => ({ children }) => <div>{children}</div>);
jest.mock('next-auth/react', () => ({
  useSession: jest.fn()
}))
import { useSession } from "next-auth/react";
const mockedUseSession = useSession as jest.Mock;


it('renders an empty wrapper if no question data is passed in', async () => {
  mockedUseSession.mockImplementation(() => ({ status: "authenticated" }));
  
  const { asFragment } = render(
    <LearnView
      video={{
        title: 'Test Video',
        description: 'Test Description',
        url: 'https://test.com',
        transcript: 'Test Transcript',
        lessonModule: 1,
        tags: { data: [] }
      }}
      setMode={(str) => str}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
