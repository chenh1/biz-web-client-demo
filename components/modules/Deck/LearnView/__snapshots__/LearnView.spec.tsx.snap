// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an empty wrapper if no question data is passed in 1`] = `
<DocumentFragment>
  <div
    class=" grid  gap-8  p-2 place-items-center w-full mt-[50%] md:mt-[20%]"
  >
    <div
      class=" grid  gap-4  p-2 w-full place-self-center place-items-center"
    >
      <h2
        class="text-night-black dark:text-max-white font-headings font-semibold text-xl sm:text-5xl text-center"
      >
        Test Video
      </h2>
      <div
        class=" grid  gap-4  p-2 justify-self-stretch sm:justify-self-center"
      >
        <button
          class=" secondary  w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-slate-700 bg-transparent border-slate-700 dark:text-max-white dark:border-max-white active:text-max-white active:bg-slate-700 active:border-slate-700 dark:active:bg-max-white dark:active:border-max-white dark:active:text-night-black px-5 py-2.5   "
          data-testid="button"
          type="button"
        >
          <div
            class="triangle left"
          >
            <div />
          </div>
          <div
            class="triangle right"
          >
            <div />
          </div>
          <span
            class="pointer-events-none relative z-10"
          >
            Skip to quiz
          </span>
        </button>
      </div>
    </div>
    <div
      class="w-full relative pt-[56.25%]"
    >
      <div
        class="absolute top-0 left-0"
        style="width: 100%; height: 100%;"
      />
    </div>
    <div
      class=" grid  gap-4  p-2 w-full"
    >
      <div
        class=" grid  gap-4  p-2 "
      >
        <div
          class="cursor-pointer flex justify-between md:justify-start"
          data-testid="accordionLabelClick"
        >
          <h4
            class="text-night-black dark:text-max-white font-headings font-semibold text-lg sm:text-md "
          >
            Transcript
          </h4>
          <div
            class="relative w-5 ml-4"
          >
            <div
              class="triangle accordion down"
            >
              <div />
            </div>
          </div>
        </div>
        <div
          class="relative overflow-hidden duration-300 ease-in-out h-[1px]"
        >
          <div
            class=" grid  gap-4  p-2 duration-300 ease-in-out dividerShouldExpand"
          >
            <div>
              Test Transcript
            </div>
            <hr
              class="w-full h-0 border-0 border-t-[1px] border-night-black-300 dark:border-max-white "
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
